sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/eventhubs_exception.hpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <azure/core/amqp/models/amqp_error.hpp>
+
+#include <stdexcept>
+#include <string>
+
+namespace Azure { namespace Messaging { namespace EventHubs {
+
+  enum class EventHubsStatusCode : int32_t
+  {
+    Invalid,
+    Ok,
+    Error,
+    Timeout,
+    Cancelled,
+  };
+
+  /**
+   * @brief An exception thrown when an EventHubs service operation fails.
+   */
+  struct EventHubsException final : std::runtime_error
```

Comment:
I see that .NET has a type like this, but I am surprised we want `<service-specific>Exception` types. If we are going to have it, like `StorageException` which was an exception, we derive from RFE there. Shouldn't this follow that pattern?

Reply:
nit: I believe we want our exception types and anything other than models and option bags to be classes rather than struct.

Reply:
RFE is inherently coupled to HTTP clients. It cannot be used outside of the context of an HTTP operation.

We've already seen customers complaining that TransportException derives from RFE, but doesn't have all the RFE fields filled in.